iocore/hostdb/HostDB.cc

```diff
@@ -1382,8 +1384,8 @@ HostDBContinuation::dnsEvent(int event, HostEnt *e)
       if (is_srv()) {
         int skip  = 0;
         char *pos = (char *)rr_data + sizeof(HostDBRoundRobin) + valid_records * sizeof(HostDBInfo);
-        SRV *q[HOST_DB_MAX_ROUND_ROBIN_INFO];
-        ink_assert(valid_records <= HOST_DB_MAX_ROUND_ROBIN_INFO);
+        std::vector<SRV *> q(valid_records);
```

Comment:
Elsewhere you have
```
HostDBInfo *infos[good]
```

Why not something similar here, as the size of `valid_records` is know? That would avoid allocating memory for the `std::vector`.